pkg/ipam/poolallocator/allocator.go

```diff
@@ -272,6 +309,63 @@ func (a *IPPoolAllocator) AllocateNext(state v1alpha2.IPAddressPhase, owner v1al
 	return ip, subnetSpec, err
 }
 
+// AllocateReserved allocates the reserved IP if exists, else allocates next
+// available IP. It returns error if pool is exausted, or in case CRD failed to update its state.
```

Comment:
exhausted
in case it fails to update IPPool's state

Reply:
Fixed.